functests/5_latency_testing/latency_testing.go

```diff
@@ -0,0 +1,221 @@
+	table.Entry("[test_id:42115] Cyclictest - Verify that the tool is working properly with valid environment variables values", getValidValuesTests(CYCLICTEST), POSITIVE_TESTING),
+	table.Entry("[test_id:42852] Cyclictest - Verify that the latency tool test should print an expected error message when passing invalid environment variables values", getNegativeTests(CYCLICTEST), NEGATIVE_TESTING),
+	table.Entry("[test_id:42849] Hwlatdetect - Verify that the tool is working properly with valid environment variables values", getValidValuesTests(HWLATDETECT), POSITIVE_TESTING),
+	table.Entry("[test_id:42856] Hwlatdetect - Verify that the latency tool test should print an expected error message when passing invalid environment variables values", getNegativeTests(HWLATDETECT), NEGATIVE_TESTING),
+)
+
+func setEnvAndGetDescription(tst latencyTest) string {
+	sb := bytes.NewBufferString("")
+	testName := tst.toolToTest
+	if tst.toolToTest == "" {
+		testName = "latency tools"
+	}
+	fmt.Fprintf(sb, "Run %s test : \n", testName)
+	nonDefaultValues := false
+	if tst.testDelay != "" {
+		setEnvWriteDescription(LATENCY_TEST_DELAY, tst.testDelay, sb, &nonDefaultValues)
+	}
+	if tst.testRun != "" {
+		setEnvWriteDescription(LATENCY_TEST_RUN, tst.testRun, sb, &nonDefaultValues)
+	}
+	if tst.testRuntime != "" {
+		setEnvWriteDescription(LATENCY_TEST_RUNTIME, tst.testRuntime, sb, &nonDefaultValues)
+	}
+	if tst.testMaxLatency != "" {
+		setEnvWriteDescription(MAXIMUM_LATENCY, tst.testMaxLatency, sb, &nonDefaultValues)
+	}
+	if tst.oslatMaxLatency != "" {
+		setEnvWriteDescription(OSLAT_MAXIMUM_LATENCY, tst.oslatMaxLatency, sb, &nonDefaultValues)
+	}
+	if tst.cyclictestMaxLatency != "" {
+		setEnvWriteDescription(CYCLICTEST_MAXIMUM_LATENCY, tst.cyclictestMaxLatency, sb, &nonDefaultValues)
+	}
+	if tst.hwlatdetectMaxLatency != "" {
+		setEnvWriteDescription(HWLATDETECT_MAXIMUM_LATENCY, tst.hwlatdetectMaxLatency, sb, &nonDefaultValues)
+	}
+	if tst.testCpus != "" {
+		setEnvWriteDescription(LATENCY_TEST_CPUS, tst.testCpus, sb, &nonDefaultValues)
+	}
+	if !nonDefaultValues {
+		fmt.Fprint(sb, "With default values of the environment variables")
+	}
+
+	return sb.String()
+}
+
+func setEnvWriteDescription(envVar string, val string, sb *bytes.Buffer, flag *bool) {
+	os.Setenv(envVar, val)
+	fmt.Fprintf(sb, "%s = %s \n", envVar, val)
+	*flag = true
+}
+
+func clearEnv() {
+	os.Unsetenv(LATENCY_TEST_DELAY)
+	os.Unsetenv(LATENCY_TEST_RUN)
+	os.Unsetenv(LATENCY_TEST_RUNTIME)
+	os.Unsetenv(MAXIMUM_LATENCY)
+	os.Unsetenv(OSLAT_MAXIMUM_LATENCY)
+	os.Unsetenv(CYCLICTEST_MAXIMUM_LATENCY)
+	os.Unsetenv(HWLATDETECT_MAXIMUM_LATENCY)
+	os.Unsetenv(LATENCY_TEST_CPUS)
+}
+
+func getValidValuesTests(toolToTest string) []latencyTest {
+	var testSet []latencyTest
+	testSet = append(testSet, latencyTest{"0", "true", "5", GUARANTEED_LATENCY, "", "", "", "3", []string{SUCCESS}, toolToTest})
```

Comment:
For more readable tests I would create the instances of latencyTest  by using a struct literal and assigning values to the fields of the struct.
Example:
`latencyTest{testDelay: "0", testRun: "true",testRuntime: "5",testMaxLatency: GUARANTEED_LATENCY, testCpus: "3", outputMsgs: []string{SUCCESS},toolToTest: toolToTest}`

Reply:
actually sounds like a very good suggestion, I agree it would be more readable. I will do the changes, Thanks!